controller/linodemachine_controller_helpers.go

```diff
@@ -74,8 +80,23 @@ func (r *LinodeMachineReconciler) newCreateConfig(ctx context.Context, machineSc
 
 		return nil, err
 	}
-	createConfig.Metadata = &linodego.InstanceMetadataOptions{
-		UserData: b64.StdEncoding.EncodeToString(bootstrapData),
+
+	region, err := machineScope.LinodeClient.GetRegion(ctx, machineScope.LinodeMachine.Spec.Region)
+	if err != nil {
+		return nil, err
+	}
+
+	if machineScope.LinodeMachine.Spec.UseStackScriptBootstrap || !slices.Contains(region.Capabilities, "Metadata") {
+		logger.Info("metadata is not enabled in this region, using StackScripts for bootstrapping")
```

Comment:
We should only log this line if the second conditional evaluates to true

Reply:
good call, let me rewrite the message to take into account both conditions